src/PowerShellEditorServices/Utility/VisitorUtils.cs

```diff
@@ -10,42 +16,336 @@ namespace Microsoft.PowerShell.EditorServices.Utility
-                    lineOffset = 0;
                 }
                 else if (astText[astOffset] == '\r')
                 {
                     // Do nothing with carriage returns... we only look for line feeds since those
                     // are used on every platform.
                 }
-                else if (!char.IsWhiteSpace(astText[astOffset]))
+                else if (astOffset >= nameStartIndex && !char.IsWhiteSpace(astText[astOffset]))
                 {
                     // This is the start of the function name so we've found our start column and line number.
                     break;
                 }
             }
 
-            return (startColumnNumber + lineOffset, startLineNumber);
+            return (startColumnNumber, startLineNumber);
+        }
+
+        /// <summary>
+        /// Calculates the start line and column of the actual function name in a function definition AST.
+        /// </summary>
+        /// <param name="functionDefinitionAst">A FunctionDefinitionAst object in the script's AST</param>
+        /// <returns>A tuple with start column and line for the function name</returns>
+        internal static (int startColumn, int startLine) GetNameStartColumnAndLineFromAst(FunctionDefinitionAst functionDefinitionAst)
+        {
+            int startOffset = functionDefinitionAst.IsFilter ? "filter".Length : functionDefinitionAst.IsWorkflow ? "workflow".Length : "function".Length;
+            return GetNameStartColumnAndLineFromAst(functionDefinitionAst, startOffset);
+        }
+
+        /// <summary>
+        /// Calculates the start line and column of the actual class/enum name in a type definition AST.
+        /// </summary>
+        /// <param name="typeDefinitionAst">A TypeDefinitionAst object in the script's AST</param>
+        /// <returns>A tuple with start column and line for the type name</returns>
+        internal static (int startColumn, int startLine) GetNameStartColumnAndLineFromAst(TypeDefinitionAst typeDefinitionAst)
+        {
+            int startOffset = typeDefinitionAst.IsEnum ? "enum".Length : "class".Length;
+            return GetNameStartColumnAndLineFromAst(typeDefinitionAst, startOffset);
+        }
+
+        /// <summary>
+        /// Calculates the start line and column of the actual method/constructor name in a function member AST.
+        /// </summary>
+        /// <param name="functionMemberAst">A FunctionMemberAst object in the script's AST</param>
+        /// <returns>A tuple with start column and line for the method/constructor name</returns>
+        internal static (int startColumn, int startLine) GetNameStartColumnAndLineFromAst(FunctionMemberAst functionMemberAst)
+        {
+            // find name index to get offset even with attributes, static, hidden ++
+            int nameStartIndex = functionMemberAst.Extent.Text.LastIndexOf(string.Concat(functionMemberAst.Name, '('), StringComparison.OrdinalIgnoreCase);
```

Comment:
Does `LastIndexOf` make sense here? Does that not include the body of the function? If it does, then recursive functions might cause some errors here. Or even something like:

```powershell
class test {
    [string] GetThing() {
        return 'GetThing('
    }
}
```

Ideally we'd be using tokens for this

Reply:
Good point, any better fix? @fflaten any ideas?

Reply:
Good catch! `IndexOf()`, which I really don't understand why I didn't use originally. 😕 
You're not allowed to use `(` before function name so shouldn't be any conflicts.